api/v1alpha1/nginxingresscontroller_types.go

```diff
@@ -62,8 +62,9 @@ type NginxIngressControllerSpec struct {
 }
 
 // DefaultSSLCertificate holds a secret in the form of a secret struct with name and namespace properties or a key vault uri
-// +kubebuilder:validation:MaxProperties=1
+// +kubebuilder:validation:MaxProperties=2
```

Comment:
leave comment describing the why and how this forces one of secret or keyvault uri